include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -182,6 +182,12 @@ class IExecutionProvider {
   */
   virtual common::Status Sync() const { return Status::OK(); }
 
+  /**
+     Set graph annotation for saving/retriving executable graphs (e.g., cuda graph).
+     Currently only CUDA execution provider supports it.
+  */
+  virtual void SetGraphAnnotation(int) {}
```

Comment:
Cheng suggests not change the interface of ep.

Reply:
> Cheng suggests not change the interface of ep.

I think RunOptions is passed in OnRunStart, you can get annotation id from run options. So it is not needed to have this function.

Reply:
will remove SetGraphAnnotation() and add run_option as an argument to ReplayGraph()